### PR TITLE
fix: NPE when `connectionCookie` is `undefined`

### PR DESCRIPTION
**Pull request by sharat87**
Fix NPE in RTS when the `connectionCookie` is `undefined`.

### CODE REVIEW

File: app/rts/src/server.ts [hunk 1]
```diff
@@ -168,49 +168,50 @@ async function tryAuth(socket: Socket) {
...
-            } catch (error) {
-                if (error.response?.status === 401) {
-                    console.info("401 received when authenticating user with cookie: " + sessionCookie);
-                } else if (error.response) {
-                    log.error("Error response received while authentication: ", error.response);
-                } else {
-                    log.error("Error authenticating", error);
-                }
-                return false;
+    if (connectionCookie == null || connectionCookie === "") {
```
Comment (IAmAnubhavSaini) [inline app/rts/src/server.ts:171]: `===` for the first check.

File: app/rts/src/server.ts [hunk 2]
```diff
@@ -168,49 +168,50 @@ async function tryAuth(socket: Socket) {
...
+        return false;
+    }
+
+    const matchedCookie = connectionCookie.match(/\bSESSION=\S+/);
```
Comment (IAmAnubhavSaini) [inline app/rts/src/server.ts:175]: Following check is missing:

```
if (!matchedCookie) {
    return false;
}
```

Then the code block in `matchedCookie` can be unindented too.

File: app/rts/src/server.ts [hunk 3]
```diff
@@ -168,49 +168,50 @@ async function tryAuth(socket: Socket) {
...
+                log.error("Error response received while authentication: ", error.response);
+            } else {
+                log.error("Error authenticating", error);
             }
+            return false;
+        }
 
-            const email = response.data.data.email;
-            const name = response.data.data.name ? response.data.data.name : email;
+        const email = response.data.data.email;
```
Comment (IAmAnubhavSaini) [inline app/rts/src/server.ts:198]: `response?.data?.data?.email`

File: app/rts/src/server.ts [hunk 4]
```diff
@@ -168,49 +168,50 @@ async function tryAuth(socket: Socket) {
 
     // const host = socket.handshake.headers.host;
     const connectionCookie = socket.handshake.headers.cookie;
```
Comment (IAmAnubhavSaini) [inline app/rts/src/server.ts:170]: `socket?.handshake?.headers?.cookie`

File: app/rts/src/server.ts [hunk 5]
```diff
@@ -168,49 +168,50 @@ async function tryAuth(socket: Socket) {
...
+            } else {
+                log.error("Error authenticating", error);
             }
+            return false;
+        }
 
-            const email = response.data.data.email;
-            const name = response.data.data.name ? response.data.data.name : email;
+        const email = response.data.data.email;
+        const name = response.data.data.name ? response.data.data.name : email;
```
Comment (IAmAnubhavSaini) [inline app/rts/src/server.ts:199]: same as above

**Review by IAmAnubhavSaini — CHANGES_REQUESTED** [review]

File: app/rts/src/server.ts (hunk 1 above)
Comment (sharat87) [reply app/rts/src/server.ts:171]: That was the problem actually. If we use `===` for `null`, then `undefined` will go through. I want to check for `null` and `undefined`. It is recommended to use `==` for null-checks in Javascript, and `===` everywhere else. Like the `smart` option in eslint: <https://eslint.org/docs/rules/eqeqeq>.

File: app/rts/src/server.ts (hunk 3 above)
Comment (sharat87) [reply app/rts/src/server.ts]: Done.

File: app/rts/src/server.ts (hunk 4 above)
Comment (sharat87) [reply app/rts/src/server.ts]: Done.

File: app/rts/src/server.ts (hunk 5 above)
Comment (sharat87) [reply app/rts/src/server.ts]: Done.

File: app/rts/src/server.ts (hunk 2 above)
Comment (sharat87) [reply app/rts/src/server.ts:175]: Good call, done. :)

File: app/rts/src/server.ts (hunk 1 above)
Comment (IAmAnubhavSaini) [reply app/rts/src/server.ts:171]: @sharat87 This change is pending. For maintainability it is very important for code to be unsurprising and boring.

Comment (sharat87) [reply app/rts/src/server.ts]: Done.

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**